Tweaks for pip mirroring at sightmachine

bremac opened the pull request:
Hi, is there anyone still out there? :-)

If so, here's a few tweaks and improvements:
1. Don't hard-code the path to python. This lets `manage-packages` run in an activated virtualenv.
2. Avoid installing binary files and wheels. The current script ignores them, and I'd rather not have platform-specific gunk on the mirror.
3. Terminate immediately if `pip install -d` fails.

bremac committed:
Support running manage-packages from a virtualenv

bremac committed:
Improvements to package download

1. Avoid pulling wheels and binary packages -- we don't want
   platform-specific gunk on the mirror.
2. Check the status of subprocess.call to make sure that pip didn't fail.

thurloat commented:
Interestingly timed pull-request @bremac, as it appears the sheepdog-assets bucket has since disappeared.

Perhaps, if you have a few extra moments it would be worth allowing `BUCKET_NAME` to be passed as a command line arg.

bremac commented:
Yeah, that's probably a good idea. Or an environment variable at least. I'll have a look tonight.

bremac committed:
Connect directly to the appropriate S3 region

This change works around large uploads to recently-created buckets failing.
This is actually a boto bug (boto/boto#2207), but since there's no sign
it will ever be fixed it's better just to work around it.

bremac committed:
Use conventional AWS configs

Read aws credentials from the many sources supported by boto, not just
environment variables. (See http://boto.cloudhackers.com/en/latest/boto_config_tut.html
for details.)

Also add some small amount of compatibility with awscli by reading from
the file specified by AWS_CONFIG_FILE, if it is set.

bremac committed:
Optionally read mirror location from environment

bremac commented:
... ooor I might forget for two months.

Anyway, I've made it optionally read the environment variables `PYPI_MIRROR_BUCKET` and `PYPI_MIRROR_DIR` to figure out where to upload to. I kept the defaults, since our internal version has our own bucket / directory hard-coded. (We use the environment variables for testing on a separate bucket.)

I also added two other changes:
- A workaround for boto/boto#2207, which causes uploads to newly-created buckets to fail; and
- The ability to use any of the [default ways to configuring boto credentials](http://boto.cloudhackers.com/en/latest/boto_config_tut.html), and to read creds from awscli configs. (Our ops guys started complaining that they had the same credentials in too many files.)

minichate commented on manage-packages:
```diff
@@ -87,8 +87,15 @@ _cached_bucket = None
 def aws_bucket():
     global _cached_bucket
```
/me is tempted to `git blame` this :/
